core/jobs/batch_jobs/mailchimp_population_jobs.py

```diff
@@ -0,0 +1,241 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Job to populate the mailchimp db with existing users."""
+
+from __future__ import annotations
+
+import ast
+
+from core import feconf
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+import mailchimp3
+from mailchimp3 import mailchimpclient
+import result
+from typing import List
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import config_models
+    from mypy_imports import user_models
+
+(config_models, user_models) = models.Registry.import_models(
+    [models.NAMES.config, models.NAMES.user])
+
+
+@staticmethod
+def _get_mailchimp_class() -> mailchimp3.MailChimp:
+    """Returns the mailchimp api class. This is separated into a separate
+    function to facilitate testing.
+
+    NOTE: No other functionalities should be added to this function.
+
+    Returns:
+        Mailchimp. A mailchimp class instance with the API key and username
+        initialized.
+    """
+
+    # The following is a class initialized in the library with the API key and
+    # username and hence cannot be tested directly. The mailchimp functions are
+    # tested with a mock class.
+    return mailchimp3.MailChimp(    # pragma: no cover
+        mc_api=feconf.MAILCHIMP_API_KEY, mc_user=feconf.MAILCHIMP_USERNAME)
+
+
+class SendBatchMailchimpRequest(beam.DoFn):
+    """DoFn to send batch mailchimp request for 500 users at a time."""
+
+    def process(
+        self, emails: List[str], batch_index_dict: int, test_run: bool
+    ) -> result.Result[str]:
+        """Add 500 users at a time, who have subscribed for newsletters,
+            to the MailChimp DB.
+
+        Args:
+            emails: list(str). List of emails of users subscribed to
+                newsletters.
+            batch_index_dict: int. Current batch index.
+            test_run: bool. Whether to use mailchimp API or not. To be set to
+                TRUE only when run from a non-production server for testing.
+
+        Raises:
+            Exception. Exception thrown by the api is raised.
+
+        Yields:
+            JobRunResult. Job run result which is either 'Ok' or an error with
+            corresponding error message.
+        """
+        emails = emails[
+            batch_index_dict * 500: (batch_index_dict + 1) * 500]
+
+        if test_run:
+            # There is a max limit of 1500 bytes for job output. Hence, only
+            # returning first and last 5 emails in batch for testing.
+            yield result.Ok(','.join(emails[: 5] + emails[-5:]))
```

Comment:
Is `emails[: 5]` also required by lint checks? If yes I think this is an issue with the lint checks.

Reply:
Yeah, adding spaces in between caused lint errors

Reply:
Can we file this as linting issue?